src/github.com/matrix-org/dendrite/clientapi/auth/storage/devices/storage.go

```diff
@@ -111,6 +115,18 @@ func (d *Database) CreateDevice(
 	return
 }
 
+// generateDeviceID creates a new device id. Returns an error if failed to generate
+// random bytes.
+func generateDeviceID() (string, error) {
```

Comment:
Why move this function from auth to here?

Reply:
Import cycles. 😖 

Should never import parent directory/package.